orttraining/orttraining/python/training/checkpoint.py

```diff
@@ -342,13 +371,76 @@ def aggregate_checkpoints(paths, pytorch_format=True):
                 _utils.state_dict_user_dict_key() in rank_state_dict:
                 state_dict[_utils.state_dict_user_dict_key()] = rank_state_dict[_utils.state_dict_user_dict_key()]
 
-    # reshape all the sharded tensors based on the original dimensions stored in sharded_states_original_dims
-    _reshape_states(sharded_states_original_dims, state_dict, loaded_mixed_precision)
+    if reshape_states:
```

Comment:
is the ```reshape_states``` flag the same as ```add_partition_info``` in the sense that these are only for intermediary states (Going from megatron+zero -> megatron (intermediary) -> single state dict? If that is the case, may we use a single flag to represent that instead of two distinct ones?
My expectation is that the final state_dict would not have the partition info subdict and would have all of its tensors reshaped as well.

Reply:
that is correct, combining into a single flag representing a partial aggregation instead